code/modules/food_and_drinks/pizzabox.dm

```diff
@@ -286,3 +287,31 @@
 	. = ..()
 	pizza = new /obj/item/reagent_containers/food/snacks/pizza/meat(src)
 	boxtag = "Meatlover's Supreme"
+
+//An anomalous pizza box that, when opened, produces the opener's favorite kind of pizza.
+/obj/item/pizzabox/anomalous
+	resistance_flags = FIRE_PROOF | LAVA_PROOF | ACID_PROOF //hard to destroy
+	can_open_on_fall = FALSE
+	var/list/pizza_types = list(/obj/item/reagent_containers/food/snacks/pizza/meat = 1,
+	/obj/item/reagent_containers/food/snacks/pizza/mushroom = 1,
+	/obj/item/reagent_containers/food/snacks/pizza/margherita = 1,
+	/obj/item/reagent_containers/food/snacks/pizza/sassysage = 0.8,
+	/obj/item/reagent_containers/food/snacks/pizza/vegetable = 0.8,
+	/obj/item/reagent_containers/food/snacks/pizza/donkpocket = 0.3,
+	/obj/item/reagent_containers/food/snacks/pizza/dank = 0.1)
+	var/static/list/pizza_preferences
+
+/obj/item/pizzabox/anomalous/Initialize()
+	. = ..()
+	if(!pizza_preferences)
+		pizza_preferences = list()
+
+/obj/item/pizzabox/anomalous/attack_self(mob/living/user)
+	QDEL_NULL(pizza)
+	if(!pizza_preferences[user.ckey])
+		pizza_preferences[user.ckey] = pickweight(pizza_types)

```

Comment:
Use `pickweight(subtypesof(/obj/item/reagent_containers/food/snacks/pizza)`

Reply:
I wanted to give each pizza type different chances to be selected due to some types like dank pizza and donkpocket pizza being very unconventional and so unlikely to be preferred by people. Is there a way to do that using the subtypesof?